### PR TITLE
ECL: access neighboring cells

### PR DESCRIPTION
**Pull request by peterrum**
Related to issue #9153.

Ping @kronbichler

### CODE REVIEW

File: include/deal.II/matrix_free/fe_evaluation.h [hunk 1]
```diff
@@ -3555,6 +3562,79 @@ FEEvaluationBase<dim, n_components_, Number, is_face, VectorizedArrayType>::
...
+             internal::MatrixFreeFunctions::DoFInfo::dof_access_cell &&
+           this->is_interior_face == false)
+    {
+      // exterior face (element-centric loop)
+      for (unsigned int i = 0; i < v_len; i++)
+        {
+          // compute actual (non vectorized) cell ID
+          const unsigned int cell_this = this->cell * v_len + i;
+          // compute face ID
+          unsigned int fn =
```
Comment (kronbichler) [inline include/deal.II/matrix_free/fe_evaluation.h]:
```suggestion
          unsigned int face_index =
```

File: include/deal.II/matrix_free/fe_evaluation.h [hunk 2]
```diff
@@ -3555,6 +3562,79 @@ FEEvaluationBase<dim, n_components_, Number, is_face, VectorizedArrayType>::
...
+      // cell or interior face face (element-centric loop)
+      for (unsigned int i = 0; i < v_len; ++i)
+        cells[i] = cell * v_len + i;
+    }
+  else if (is_face &&
+           this->dof_access_index ==
+             internal::MatrixFreeFunctions::DoFInfo::dof_access_cell &&
+           this->is_interior_face == false)
+    {
+      // exterior face (element-centric loop)
```
Comment (kronbichler) [inline include/deal.II/matrix_free/fe_evaluation.h]:
```suggestion
      // exterior face (element-centric loop): for this case, we need to
      // look into the FaceInfo field that collects information from both
      // sides of a face once for the global mesh, and pick the face id that
      // is not the local one (cell_this).
```

**Comment by peterrum** [comment]
Done!

**Comment by peterrum** [comment]
@kronbichler Once you merge this, I will rebase #9155.

**Pull request merged**